src/data/polynomial/ring_division.lean

```diff
@@ -317,6 +317,17 @@ begin
   simp only [mem_roots hp, multiset.mem_to_finset, set.mem_set_of_eq, finset.mem_coe]
 end
 
+lemma exists_max_root [linear_order R] (p : polynomial R) (hp : p ≠ 0) :
+  ∃ x₀, ∀ x, p.is_root x → x ≤ x₀ :=
```

Comment:
You really want to use the following lemma in the proof. Could you factor out this lemma and put it in `data.set.finite`?
```lean
theorem set.exists_upper_bound_image {α β : Type*} [nonempty α] [linear_order β] (s : set α) (f : α → β) (h1 : s.finite) : 
  ∃ (a : α), ∀ (b : α), b ∈ s → f b ≤ f a
```